src/NuGet.Core/NuGet.PackageManagement/BuildIntegration/DependencyGraphRestoreUtility.cs

```diff
@@ -369,5 +405,14 @@ public static async Task<PackageSpec> GetProjectSpec(IDependencyGraphProject pro
 
             return restoreContext;
         }
+
+        private class ProjectRestoreSpec
+        {
+            public DependencyGraphSpec DgSpec { get; set; }
```

Comment:
I'd prefer that all the setters were removed and that this type guarantees that all properties are non-null.  You can move property initializers into this class.

Reply:
Made them readonly properties and only sets value from constructor now.